### DISCUSSION TITLE
Dokku ps:rebuild failed after an upgrade of dokku

### DISCUSSION THREAD

**Discussion by eager-2-learn** [dokku/dokku#4977] [General]
After the dokku upgrade the "ps:rebuild" failed on "Fetching .debs for libonig4"
How can I solve this problem?

$ dokku version
dokku version 0.24.10
$ dokku ps:rebuild mportal
...
-----> Fetching .debs for libonig-dev
       Reading package lists...
       Building dependency tree...
       0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 102 not upgraded.
       Need to get 0 B/49.0 kB of archives.
       After this operation, 0 B of additional disk space will be used.
       Download complete and in download only mode
-----> Fetching .debs for libonig4
       Reading package lists...
       Building dependency tree...
       0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 102 not upgraded.
       Need to get 0 B/119 kB of archives.
       After this operation, 0 B of additional disk space will be used.
       Download complete and in download only mode
-----> Installing libonig-dev_6.7.0-1_amd64.deb
-----> Installing libonig4_6.7.0-1_amd64.deb

$ apt policy dokku
dokku:
  Installed: 0.24.10
  Candidate: 0.26.6
$ apt-get install dokku
$ dokku version 0.26.6
$ dokku ps:rebuild mportal
...
-----> Fetching .debs for libonig-dev
       Reading package lists...
       Building dependency tree...
       0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 12 not upgraded.
       Need to get 54.2 kB of archives.
       After this operation, 0 B of additional disk space will be used.
       Get:1 http://archive.ubuntu.com/ubuntu focal/universe amd64 libonig-dev amd64 6.9.4-1 [54.2 kB]
       Fetched 54.2 kB in 0s (811 kB/s)
       Download complete and in download only mode
-----> Fetching .debs for libonig4
       Reading package lists...
       Building dependency tree...
E: Unable to locate package libonig4
2021/12/16 12:41:14 exit status 100
2021/12/16 12:41:14 exit status 1
 !     exit status 1

**Comment by josegonzalez** [comment]
I think you maybe updated herokuish to the heroku-20 (focal), and libonig4 doesn't exist for that (libonig does).

Can you show the output of `dokku report`? I'm assuming you have a plugin that is installing this, and you may wish to update that plugin (or switch to dokku-apt and specify the new version in an aptfile).

**Comment by eager-2-learn** [comment]
I use herokuish version 0.5.28
That is herokuish-18 (bionic).

  dokku enter mportal
  root@4c639d168ab8:/# uname -r
  4.15.0-163-generic
  root@4c639d168ab8:/# cat /etc/issue
  Ubuntu 18.04.5 LTS \n \l
  root@4c639d168ab8:/# dpkg --get-selections | grep libonig
  libonig-dev			   		        install
  libonig4:amd64					install

As you see libonig4 is still installed.

[dokku_report.txt](https://github.com/dokku/dokku/files/7792244/dokku_report.txt)

I installed the dokku apt plugin
$ dokku plugin:list | grep apt 
  apt                  0.12.0 enabled    Inject deb packages into dokku based on files in project
But I do not know how to use that apt plugin.
